membership_product_set/models/res_partner.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Yu Weng <yweng@elegosoft.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.multi
+    def create_membership_invoice(self, product_id=None, datas=None):
+        product_id = product_id or datas.get('membership_product_id')
+        amount = datas.get('amount', 0.0)
+        product = self.env['product.product'].browse(product_id)
+        invoice_list = []
+        if product.membership_set:
+            for partner in self:
+                addr = partner.address_get(['invoice'])
+                if partner.free_member:
+                    raise UserError(_("Partner is a free Member."))
+                w = _("Partner doesn't have an address to make the invoice.")
+                if not addr.get('invoice', False):
+                    raise UserError(w)
+                account_id = partner.property_account_receivable_id.id
+                position_id = partner.property_account_position_id.id
+                invoice = self.env['account.invoice'].create({
```

Comment:
Add Condition `if  account_id and position_id`
Because of Sometime property_account_receivable_id  & property_account_position_id is Deleted then ..

Reply:
Thanks for the review. I have added the condition "if account_id".
"fiscal_position_id" is not required for account.invoice. So we can just ignore it, if it was not set.